src/service/topics.ts

```diff
@@ -128,6 +128,17 @@ const getLatestTopicPost = async (topicName: string): Promise<TopicLatestPostDat
   }
 };
 
+const verifyCommunityName = async (name) => {
+  try {
+    const resApi = await api.get(`/topics/is-exist?name=${name}`);
+    console.warn('resApi', JSON.stringify(resApi.data));
+    return resApi.data;
+  } catch (error) {
+    crashlytics().recordError(new Error(error));
+    return error.response.data;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `verifyCommunityName` function is making an API call to check if a community name exists. However, it's directly interpolating the `name` parameter into the URL which could lead to security vulnerabilities such as URL Injection if not properly sanitized. It would be better to use query parameters provided by the axios library (which your `api` object seems to be an instance of) to safely include the `name` in the request.

Also, logging the entire response with `console.warn` on line 134 might expose sensitive information in a production environment. Consider removing this or limiting the logged information.

Lastly, returning the error response data directly might not be the best approach. It would be better to throw an error and handle it in the calling function. This way, you can differentiate between successful and erroneous responses more clearly.

Here's how you can improve this:

```diff
- const verifyCommunityName = async (name) => {
+ const verifyCommunityName = async (name) => {
    try {
-     const resApi = await api.get(`/topics/is-exist?name=${name}`);
+     const resApi = await api.get('/topics/is-exist', { params: { name } });
-     console.warn('resApi', JSON.stringify(resApi.data));
      return resApi.data;
    } catch (error) {
      crashlytics().recordError(new Error(error));
-     return error.response.data;
+     throw error;
    }
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->